### DISCUSSION TITLE
🚑 Detox, CircleCI and Android are not compatible

### DISCUSSION THREAD

**Discussion by a-tokyo** [wix/Detox#3704] [Q&A]
### Although Detox is supposedly [compatible with CircleCI](https://github.com/wix/Detox#about), It is impossible to run Android on CircleCI due to a requirements deadlock. 

- Detox only supports macOS. https://github.com/wix/Detox#environment
- CircleCI MacOS machines come with Java 11. Which is supported by android command line tools https://developer.android.com/studio/ (refer to Command line tools only section for downloads for CI)
- CircleCI MacOS machines don't support running android emulator with Java 11 and latest command line tools due to Incompatible hardware.  (Note, a common thought would be to go with Intel HAXM package installation, but it's not supported due to the hardware as well). [CircleCI build hosts are Intel-based Mac](https://circleci.com/docs/using-macos/#apple-silicon-support).
 ```
 Android Emulator requires an Intel processor with VT-x and NX support.
 ```

**Comment by noomorph** [comment]
We support Linux and Windows as well. Maybe we are a bit slower and less
concentrated on fixing OS-specific issues, but still we do support all the
three major platforms.

пн, 14 лист. 2022 р. о 13:14 Ahmed Tarek ***@***.***> пише:

**Reply by noomorph** [reply]
I updated our README to eliminate the source of the confusion.

**Comment by d4vidi** [comment]
@A-Tokyo thanks for bringing this to our attention.
Mind that it's **not that Detox support only MacOS**, as @noomorph explained -- it's Apple that only allow for running iOS simulators on Mac machines. **Android can be launched in any of the mentioned environments**.
I'm not familiar with the specs of CircleCI but strikes me odd that they do not support Android emulators. Perhaps running in _headless_ mode or using a non-default gpu setting would help.

https://wix.github.io/Detox/docs/cli/test#options
https://developer.android.com/studio/run/emulator-acceleration#accel-graphics (arguments are supported by detox via the `--gpu` switch)
